File: p6.pl
#!/usr/bin/perl

use strict;
use Data::Dumper;

my $input = 'i6.txt';

open(IN, "$input") || die "cant read $input: $!\n";

my $pkt = 0;
my $marker = '';

while (<IN>) {
	chomp;
	my $line = $_;
	for my $l (4,14) {
		my $pos = lookfor($line,$l);
		if ($pos) {
			printf ("marker of lenght %i: \"%s\" found at %i\n", $l, substr($line,$pos,$l), $pos + $l );
		}
	}
}

sub lookfor () {
	my $str = shift;
	my $len = shift;
	my $pkt = 0;
	my $found = 0;
	my $marker = '';
	while ($pkt < length($str)-$len){
		$marker = substr($str, $pkt++, $len);
		if (ismarker($marker)) {
			$found = $pkt-1;
			last;
		} else {
			next;
		}
	}
	return $found;
}

sub ismarker () {
	my $mark = shift;
	my $last = '';
	foreach (sort(split //, $mark)) {
		if ( $_ eq $last ) {
			return 0;
		} else {
			$last = $_;
		}
	}
	return 1;
}

File: p6.py
#!/usr/bin/python3

with open("i6.txt", "r") as fd:
    line = fd.read()

def getn(string,n):
	getn = None
	for i in range(len(string)):
    		window = string[i:i+n]
    		if len(window) == len(set(window)) and not getn:
        		getn = i + n
        		break
	return getn
			
print("Part 1:", getn(line,4))
print("Part 2:", getn(line,14))
